### PR TITLE
Fix h2 spec

### PR DESCRIPTION
**Pull request by ndushay**
## Why was this change made?

h2 spec never succeeded due to introduced error

### CODE REVIEW

File: spec/features/create_object_h2_spec.rb [hunk 1]
```diff
@@ -35,11 +38,11 @@
     expect(page).to have_content(collection_title)
     expect(page).to have_content('+ Deposit to this collection')
 
-    item_id = page.current_url.split('/').last
+    collection_id = page.current_url.split('/').last
```
Comment (ndushay) [inline spec/features/create_object_h2_spec.rb:41]: this was confusing me for a bit

File: spec/features/create_object_h2_spec.rb [hunk 2]
```diff
@@ -59,7 +62,6 @@
     click_button 'Continue'
 
     # Work Deposit view
-    fill_in 'What\'s changing?', with: 'Initial metadata'
```
Comment (ndushay) [inline spec/features/create_object_h2_spec.rb:62]: this was a bug introduced -- the PR for reserve a purl was old, so I guess this changed.

File: spec/features/create_object_h2_spec.rb [hunk 3]
```diff
@@ -6,17 +6,20 @@
...
   end
 
   scenario do
+    # remove modal for deposit in progress, if present
     click_button 'No' if page.has_content?('Continue your deposit', wait: Settings.post_authentication_text_timeout)
-    click_link '+ Create a new collection'
 
-    # Checks for specific content on create collection view
+    # CREATE COLLECTION
+    expect(page).to have_content('+ Create a new collection')
```
Comment (jcoyne) [inline spec/features/create_object_h2_spec.rb:17]: This is unnecessary duplication. Is there a reason for it?

File: spec/features/create_object_h2_spec.rb (hunk 2 above)
Comment (jcoyne) [reply spec/features/create_object_h2_spec.rb:62]: Yeah, this is no longer shown on the first version

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: spec/features/create_object_h2_spec.rb (hunk 3 above)
Comment (ndushay) [reply spec/features/create_object_h2_spec.rb:17]: i can try without;  there was a lot of debugging going on with what came before.

Comment (ndushay) [reply spec/features/create_object_h2_spec.rb]: yup - it's fine without.  w00t!  Maybe it's more robust now.  I hope so.

**Comment by ndushay** [comment]
@jcoyne ready for re-review.  Thanks for the suggestion.

**Review by jcoyne — APPROVED** [review]

**Pull request merged**